use_cases/eluc/predictors/evaluation/evaluation.py

```diff
@@ -0,0 +1,92 @@
+"""
+Class to evaluate predictors given a config on a dataset.
+Also a script to demo how it works.
+"""
+import importlib
+import json
+from pathlib import Path
+
+import pandas as pd
+
+import data.constants as constants
+from data.eluc_data import ELUCData
+from persistence.persistors.hf_persistor import HuggingFacePersistor
+from predictors.predictor import Predictor
+from predictors.evaluation.validator import Validator
+
+class Evaluator():
+    """
+    Evaluator class to evaluate predictors on a dataset.
+    Uses a config to dynamically load predictors.
+    The config must point to the classpath of a serializer that can call .load() to return a Predictor object.
+    Alternatively, it may use a HuggingFace url to download a model to a given path, THEN load with the serializer.
+    """
+    def __init__(self, config: dict):
+        """
+        Initializes the Evaluator with the custom classes it has to load.
+        """
+        self.predictors = self.dynamically_load_models(config)
+        # We don't pass change into the outcomes column.
+        self.validator = Validator(constants.CAO_MAPPING["context"], constants.CAO_MAPPING["actions"], ["ELUC"])
+
+    def dynamically_load_models(self, config: dict) -> list[Predictor]:
+        """
+        Uses importlib to dynamically load models from a config.
+        Config must have a list of models with the following:
+            - type: "hf" or "local" to determine if it is a HuggingFace model or local model.
+            - name: name of the serializer class to load.
+            - classpath: path to the class that calls .load()
+            - filepath: path to the model on disk or where to save the HuggingFace model.
+            - (optional) url: url to download the model from HuggingFace.
+        Returns a dict with keys being the filepath and values being the Predictor object.
+        """
+        predictors = {}
+        for model in config["models"]:
+            # We dynamically instantiate model_instance as some sort of class that can handle .load() and returns
+            # a Predictor object.
+            spec = importlib.util.spec_from_file_location(model["name"], model["classpath"])
+            module = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(module)
+            model_instance = getattr(module, model["name"])
```

Comment:
Dynamic load function uses importlib to load arbitrary classes. Did not realize python could do this!